internal/generate/packagemanifest/packagemanifest_test.go

```diff
@@ -0,0 +1,272 @@
+		})
+
+		Context("to update an existing package manifest file", func() {
+			It("should return a file with one updated channel CSV name", func() {
+				g = Generator{
+					OperatorName: operatorName,
+					Version:      "0.0.2",
+					ChannelName:  "alpha",
+					getBase:      makeBaseGetter(pkgManDefaultContent),
+				}
+				pkg, err := g.generate()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(pkg).To(Equal(&apimanifests.PackageManifest{
+					Channels: []apimanifests.PackageChannel{
+						{Name: "alpha", CurrentCSVName: genutil.MakeCSVName(operatorName, "0.0.2")},
+					},
+					DefaultChannelName: "alpha",
+					PackageName:        operatorName,
+				}))
+			})
+			It("should return a file with two channels", func() {
+				g = Generator{
+					OperatorName: operatorName,
+					Version:      "0.0.2",
+					ChannelName:  "stable",
+					getBase:      makeBaseGetter(pkgManDefaultContent),
+				}
+				pkg, err := g.generate()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(pkg).To(Equal(&apimanifests.PackageManifest{
+					Channels: []apimanifests.PackageChannel{
+						{Name: "alpha", CurrentCSVName: genutil.MakeCSVName(operatorName, version)},
+						{Name: "stable", CurrentCSVName: genutil.MakeCSVName(operatorName, "0.0.2")},
+					},
+					DefaultChannelName: "alpha",
+					PackageName:        operatorName,
+				}))
+			})
+			It("should return a file with two channels and an updated default channel", func() {
+				g = Generator{
+					OperatorName:     operatorName,
+					Version:          "0.0.2",
+					ChannelName:      "stable",
+					IsDefaultChannel: true,
+					getBase:          makeBaseGetter(pkgManDefaultContent),
+				}
+				pkg, err := g.generate()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(pkg).To(Equal(&apimanifests.PackageManifest{
+					Channels: []apimanifests.PackageChannel{
+						{Name: "alpha", CurrentCSVName: genutil.MakeCSVName(operatorName, version)},
+						{Name: "stable", CurrentCSVName: genutil.MakeCSVName(operatorName, "0.0.2")},
+					},
+					DefaultChannelName: "stable",
+					PackageName:        operatorName,
+				}))
+			})
+		})
+
+	})
+
+})
+
+func makeBaseGetter(content string) getBaseFunc {
+	return func() (*apimanifests.PackageManifest, error) {
+		return marshalContent(content)
+	}
+}
+
+func initTestPackageManifestsHelper() {
+	var err error
+	pkgManDefault, err = marshalContent(pkgManDefaultContent)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+	pkgManSingleChannel, err = marshalContent(pkgManSingleChannelContent)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+}
+
+func readFileHelper(path string) []byte {
+	b, err := ioutil.ReadFile(path)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+	return b
+}
+
+func marshalContent(content string) (*apimanifests.PackageManifest, error) {
+	base := &apimanifests.PackageManifest{}
+	if content == "" {
+		return base, nil
+	}
+	err := yaml.Unmarshal([]byte(content), base)
+	return base, err
+}
+
+const pkgManDefaultContent = `channels:
```

Comment:
Is there a reason these are declared as const here instead of just normal vars in the outermost BeforeEach()?

Reply:
I don't see a good reason to re-define constant strings that won't change.

Reply:
I just meant here at the bottom instead of the top. just makes it weird to read through because they're referenced "before" they're defined. And if you put them in the outermost before each, you would still only be defining them once.

Reply:
Ah I see. I'm fine with that.